docs/workflows/copy.rst

```diff
@@ -1,43 +1,189 @@
-
-
-.. _copy-workflow:
-
-Copy workflow
--------------
-
-You can use the modify endpoint on a repository to copy all content present in one repository
-version to another by specifying a base version.
-
-.. literalinclude:: ../_scripts/copy.sh
-   :language: bash
-
-Repository Version GET response (after task complete):
-
-.. code:: json
-
-    {
-        "base_version": "/pulp/api/v3/repositories/rpm/rpm/c8c632eb-cf9a-494f-8e12-b3e651b6e75e/versions/1/",
-        "content_summary": {
-            "added": {},
-            "present": {
-                "rpm.package": {
-                    "count": 1,
-                    "href": "/pulp/api/v3/content/rpm/packages/?repository_version=/pulp/api/v3/repositories/rpm/rpm/c8c632eb-cf9a-494f-8e12-b3e651b6e75e/versions/3/"
-                }
-            },
-            "removed": {
-                "rpm.advisory": {
-                    "count": 1,
-                    "href": "/pulp/api/v3/content/rpm/advisories/?repository_version_removed=/pulp/api/v3/repositories/rpm/rpm/c8c632eb-cf9a-494f-8e12-b3e651b6e75e/versions/3/"
-                }
-            }
-        },
-        "number": 3,
-        "pulp_created": "2019-11-27T16:07:21.561748Z",
-        "pulp_href": "/pulp/api/v3/repositories/rpm/rpm/c8c632eb-cf9a-494f-8e12-b3e651b6e75e/versions/3/"
-    }
+If you want to copy RPM content from one repository into another repository, you have two options
+for doing so.
 
+
+.. _basic-modify-workflow:
+
+Basic Repository Modification API
+---------------------------------
+
+Like all Pulp repositories, you can use the ${repo_href}/modify/ endpoint to:
+
+* add or remove individual content units from a repository by HREF
+* roll back the content present in a repository to that of a previous version using 'base_version'
+* clone a repository version from another repo using 'base_version'
+
+.. code-block:: bash
+
+    export CONTENT_HHREF=/pulp/api/v3/content/rpm/packages/9ee09de7-5fff-4805-8b30-9ab95493317d/
+
+    # Add content units to the repository
+    http POST $BASE_ADDR$REPO_HREF'modify'/ add_content_units:=["$CONTENT_HREF"]
+
+    # Remove content units from the repository
+    http POST $BASE_ADDR$REPO_HREF'modify'/ remove_content_units:=["$CONTENT_HREF"]
+
+    # Clone a repository (can be composed with addition or removal of units)
+    export TASK_URL=$(http POST $BASE_ADDR$REPO_HREF'modify'/ \
+        base_version=$REPOVERSION_HREF_WITH_PKG | jq -r '.task')
+
+    # Poll the task (here we use a function defined in docs/_scripts/base.sh)
+    wait_until_task_finished $BASE_ADDR$TASK_URL
+
+    # After the task is complete, it gives us a new repository version
+    echo "Set REPOVERSION_HREF from finished task."
+    export REPOVERSION_HREF=$(http $BASE_ADDR$TASK_URL| jq -r '.created_resources | first')
+
+    echo "Inspecting RepositoryVersion."
+    http $BASE_ADDR$REPOVERSION_HREF
+
+
+.. copy-workflow:
+
+Advanced copy workflow
```

Comment:
Is it worth specifying config structure in the docs with the short description of each parameter there? Recipes will help to understand how to use them, however having a list of thing that you can possibly put into config might be useful. What do you think?

Reply:
Eventually, I think so, but since there is only 5 of them at the moment and 3 of those are mandatory, I'm not sure if it's worth it for now.